codex/sales/states/preparing.nim

```diff
@@ -78,7 +78,16 @@ method run*(state: SalePreparing, machine: Machine): Future[?State] {.async.} =
     request.id,
     data.slotIndex
   ), error:
+    trace "Creation of reservation failed"
+    # Race condition:
+    # reservations.findAvailability (line 64) is no guarantee. You can never know for certain that the reservation can be created until after you have it.
+    # Should createReservation fail because there's no space, we proceed to SaleIgnored.
+    if error of BytesOutOfBoundsError:
+      return some State(SaleIgnored())
```

Comment:
We should add the metric and log here, however again, I don't think it's bad if we hit this exception. The slot will go back in the queue and it'll get re-processed when more availability is added.